src/repr/src/row.rs

```diff
@@ -895,6 +895,13 @@ impl Row {
         row.packer().extend(slice.iter());
         row
     }
+
+    /// Returns the total amount of bytes used by this row.
+    pub fn byte_len(&self) -> usize {
+        let heap_size = self.data.len().saturating_sub(self.data.inline_size());
+        let inline_size = std::mem::size_of::<Self>();
+        inline_size.saturating_add(heap_size)
```

Comment:
This doesn't seem correct. If a row contains, say 25 bytes, it will be size of self (32) plus 25. The above implementation does:
1. `heap_size` = 25 - 24 -> 1
2. `inline_size` = 32
3. `inline_size + heap_size` -> 33, but the actual consumed memory is 32+25.

It's late and I might be wrong, but I think this should be clarified. I might be missing something and assume wrong sizes, happy to stand corrected!